src/utils/validation/registration/referenceValidation.ts

```diff
@@ -304,8 +304,22 @@ export const reportReference = baseReference.shape({
 });
 
 // Degree
+const unconfirmedDocumentSchema = Yup.object({
+  type: Yup.string(),
+  text: Yup.string().nonNullable().required(resourceErrorMessage.typeRequired),
+});
+
+const confirmedDocumentSchema = Yup.object({
+  type: Yup.string(),
+  identifier: Yup.string().url().required(),
+  sequence: Yup.number().required(),
+});
+
 const degreePublicationInstance = Yup.object<YupShape<DegreePublicationInstance>>({
   type: Yup.string().oneOf(Object.values(DegreeType)).required(resourceErrorMessage.typeRequired),
+  related: Yup.array().of(
+    Yup.lazy((item) => (item.type === 'UnconfirmedDocument' ? unconfirmedDocumentSchema : confirmedDocumentSchema))
```

Comment:
Har ikke vært borti lazy her før, så usikker på hvordan den virker i praksis.
Tidligere har vi mange `when()`-sjekker som sjekker bare når det gjelder for enkelte type. Denne skal vel valideres bare når `type = PhD`, så tenker det er en naturlig tilnærming?